packages/compass-components/src/components/links/link.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useContext, forwardRef } from 'react';
+import { Link as LGLink } from '@leafygreen-ui/typography';
+import LGButton from '@leafygreen-ui/button';
+import LGIconButton from '@leafygreen-ui/icon-button';
+
+type LinkContextValue = {
+  utmSource?: string;
+  utmMedium?: string;
+};
+const LinkContext = React.createContext<LinkContextValue>({});
+export const LinkProvider: React.FC<LinkContextValue> = ({
+  utmSource,
+  utmMedium,
+  children,
+}) => {
+  const value: LinkContextValue = {
+    utmSource,
+    utmMedium,
+  };
+
+  return <LinkContext.Provider value={value}>{children}</LinkContext.Provider>;
+};
+
+// exported for testing purposes only
+export const EXCLUDED_MONGODB_HOSTS = [
+  'compass-maps.mongodb.com',
+  'evergreen.mongodb.com',
+  'downloads.mongodb.com',
+  'cloud.mongodb.com',
+];
```

Comment:
I mentioned this elsewhere, now that this is only applied to UI links rendered with certain components, we probably can remove this list of hosts, it's not really needed as we are not intercepting requests anymore